src/pybind/mgr/mgr_util.py

```diff
@@ -627,63 +627,48 @@ def create_self_signed_cert(organisation: str = 'Ceph',
 
     """
 
-    from OpenSSL import crypto
-    from uuid import uuid4
-
     # RDN = Relative Distinguished Name
     valid_RDN_list = ['C', 'ST', 'L', 'O', 'OU', 'CN', 'emailAddress']
 
-    # create a key pair
-    pkey = crypto.PKey()
-    pkey.generate_key(crypto.TYPE_RSA, 2048)
-
-    # Create a "subject" object
-    req = crypto.X509Req()
-    subj = req.get_subject()
-
     if dname:
         # dname received, so check it contains valid RDNs
         if not all(field in valid_RDN_list for field in dname):
             raise ValueError("Invalid DNAME received. Valid DNAME fields are {}".format(', '.join(valid_RDN_list)))
     else:
         dname = {"O": organisation, "CN": common_name}
 
-    # populate the subject with the dname settings
-    for k, v in dname.items():
-        setattr(subj, k, v)
+    import json
+    import subprocess
 
-    # create a self-signed cert
-    cert = crypto.X509()
-    cert.set_subject(req.get_subject())
-    cert.set_serial_number(int(uuid4()))
-    cert.gmtime_adj_notBefore(0)
-    cert.gmtime_adj_notAfter(10 * 365 * 24 * 60 * 60)  # 10 years
-    cert.set_issuer(cert.get_subject())
-    cert.set_pubkey(pkey)
-    cert.sign(pkey, 'sha512')
+    result = subprocess.run(["python3", "-m", "ceph.pybind.mgr.create_self_signed_cert",
+                             base64.b64encode(json.dumps(dname).encode("utf-8"))],
+                            capture_output=True, text=True)
 
-    cert = crypto.dump_certificate(crypto.FILETYPE_PEM, cert)
-    pkey = crypto.dump_privatekey(crypto.FILETYPE_PEM, pkey)
+    # Check result with a CompletedProcess
+    if result.returncode != 0 or result.stderr != '':
+        raise ValueError(result.stderr)
 
-    return cert.decode('utf-8'), pkey.decode('utf-8')
+    cert, pkey = result.stdout.strip().split("#######")
```

Comment:
Why not creating 2 different calls, one for generating the Pkey, and another for the cert (passing the key as input arg)?

Reply:
Done.